# Ok let's see what we can do in the way of data manipulation...

In [61]:
import numpy as np
import pandas as pd
import os

## GM Datasets

In [62]:
os.listdir("GM Datasets/")

['Input_Cost%2C+Location.xlsx',
 'Problem_VehicleShipmentRequirement.csv',
 'Problem_VehicleShipmentRequirement.zip']

OK, so the General Motors Datasets has two tablets of information, one pertaining to Vehicle Shipping Requirements and the other to Input Cost and Location.

### Vehicle Shipping Requirements

In [63]:
df_ship = pd.read_csv("GM Datasets/Problem_VehicleShipmentRequirement.csv")

In [64]:
len(df_ship)

1866687

In [65]:
df_ship

VIN Plant  Dealer  \
0        000003CD-DD91-4EAF-8CDB-D31A35CB4C65    RO   14135   
1        00000686-CBF0-455B-BDA8-85EAADCC31B9    RO   13449   
2        000017BF-BDF6-4B0B-88D5-46823B40DC1B    RO   18672   
3        00001999-A906-4391-A4DC-D4AC235EA6A3    FF   13619   
4        00001EDD-CDE4-49EB-957A-6FB1FE093E80    RO   16070   
5        0000209E-135A-4EB7-B3CF-8B139BA3A24F    3A   13782   
6        0000253D-FB0F-4EB9-8FBB-F3B5F91C20C4    RO   16070   
7        000025D5-B3E3-40F2-B64E-FC0D92FE607A    3A   16263   
8        0000306A-CBB7-449C-A144-AC6EFEA1F8F3    FF   16017   
9        00003411-7D48-443E-93B9-47C9B6BABD05    SO   14629   
10       00003688-43CA-4A17-9C9C-386D131630C1    SO   13752   
11       00005A68-41A6-427E-ADF6-C56EAF93CF7B    3A   15716   
12       00006047-F9A6-4F79-9ADF-794057EB62E9    RO    9415   
13       00007349-8D8D-4372-AE48-8C39579BCDB4    RO   15265   
14       0000740E-C9F5-4E56-9F49-1CCAFAF0B6C1    SO   13914   
15       00007626-5DF1-40FB-9A5F-D06064A51320    RO   15801   
16       00008602-56A6-4B65-803C-2680FAD31EFB    RO   15534   
17       00008C21-6A2B-4890-8A97-3002652B60A3    3A   13196   
18       00008E79-F383-480C-9403-97DFEA64A593    SO   15279   
19       000097F2-E1B4-408E-BBDA-70577F9351AD    FF   14797   
20       00009DB2-8646-41F4-90DB-E446444A556D    RO   20369   
21       0000A295-4E7F-4BEE-80F5-941C78AFE491    FF   15535   
22       0000AE2C-DFFA-4CC8-B59D-C70E003FACDE    SO   15966   
23       0000B87E-5F06-48EA-90C1-128538F306A7    RO   14666   
24       0000D7F9-DCB5-4CEC-ADF1-3763367A5DF2    RO   13577   
25       0000DF31-1E60-404E-931D-403D0AA39673    3A   16214   
26       0000E5BB-5C79-4EF2-BACC-CB32864F8006    SO   20570   
27       0000E801-8446-4EE9-B60F-02F964F0D5DA    3A   13777   
28       0000EC58-BA13-424A-A63B-D7DF7A710C84    SO   20625   
29       000100F2-6A46-4F7A-8790-D7F0C8F2BE69    3A   20783   
...                                       ...   ...     ...   
1866657  FFFF52F6-708E-43E8-A270-1D48CB40A465    RO   16147   
1866658  FFFF5AAF-B213-4BC6-9E15-56EDD331F2AD    SO   19182   
1866659  FFFF5F8B-146E-463F-946C-0851A6B815DC    FF    8701   
1866660  FFFF615F-B106-41E8-90EF-30AC0AE9A5CC    3A   11991   
1866661  FFFF619C-C6C9-48B7-B139-AE018A4F436A    SO   19313   
1866662  FFFF6B78-E878-4891-85EE-5B97BA9A5215    SO   15956   
1866663  FFFF6DA1-3628-49D1-B65B-D9D9EE93481A    RO   15521   
1866664  FFFF6FEE-E2AC-4AD3-8CD1-E597A5D0354F    3A   17937   
1866665  FFFF7545-74C3-4F79-8CCB-6BA2E5677215    RO   13265   
1866666  FFFF79B6-D596-487F-AFFB-D22C1846AFF5    3A   15647   
1866667  FFFF7BBC-18B5-4493-A979-C1B428AE037A    3A   16760   
1866668  FFFF9105-C26E-4AC2-A782-2B34B490170A    FF   14604   
1866669  FFFF91B8-2D5A-446B-82B6-E08B82220AB2    FF   13220   
1866670  FFFF943A-694C-40AC-AAC3-F4AD14631B65    RO   18770   
1866671  FFFF9BEE-3C4D-406B-92B1-9A96F51347B0    RO   19244   
1866672  FFFF9CD7-62CD-418D-8DBD-12D7B989A3BE    RO   14518   
1866673  FFFF9D75-BE5E-4730-AFF2-7B997F0C9693    RO    9213   
1866674  FFFFA905-0DEC-4215-B2BB-09D9D339FBD3    RO   18674   
1866675  FFFFABC1-FEDB-462D-83E3-3C0F6D54FBF8    RO   16092   
1866676  FFFFAD80-181E-4BAC-8DF4-8865FAB5CEEA    SO   20710   
1866677  FFFFC36C-02DB-425F-8C74-873C4FA657F6    RO   15606   
1866678  FFFFC3B4-A93E-43AC-BF83-83383B973681    SO   19412   
1866679  FFFFC5F9-A605-464B-ACE0-083146B17AE3    3A   20891   
1866680  FFFFC95D-D212-4E19-9A69-8942FEF253CC    SO   13319   
1866681  FFFFCBB8-4E30-4898-8FBB-85C3E9A11207    FF   14716   
1866682  FFFFDA87-FB19-4C91-9F50-623F58377F92    SO   13788   
1866683  FFFFDC17-10BB-42C5-AD72-FEAD19629F73    3A   19973   
1866684  FFFFDD42-52C3-4B23-B7E9-33F464374949    3A   13132   
1866685  FFFFE7FF-AA89-450F-97F6-BCFC6D28508E    FF   19962   
1866686  FFFFF4F1-65C3-4F86-89CC-5269053DF9BA    RO   19274   

          Plant_Arrival_Time  
0           3/8/2015 1:53:00  
1          3/16/2015 2:58:00  
2          3/31/2015 6:17:00  
3        10

In [66]:
df_ship["Plant"].value_counts()

SO    583777
RO    555245
3A    422981
FF    304684
Name: Plant, dtype: int64

We've got a four plants, with a lot of cars being shipped from each.  But we observe that the highest turnover, 'SO', has a few hundred thousand more than the lowest, 'FF'.  Could be pertinent

In [67]:
len(df_ship["Plant"].unique())

4

Obviously.

In [68]:
df_ship["Dealer"].value_counts()

15673    6252
13695    4972
15000    4384
13788    4018
9109     3933
15995    3904
13449    3798
15675    3679
20931    3457
15289    3294
18429    3174
14797    3152
13800    3132
16194    3096
18422    2935
15716    2812
15652    2776
18435    2596
13757    2587
14371    2576
15699    2554
16013    2537
20239    2429
16263    2416
15638    2384
13429    2328
14739    2305
15637    2258
13875    2250
16268    2233
         ... 
8938        1
12685       1
15759       1
17781       1
17236       1
15943       1
20520       1
9060        1
8772        1
17239       1
9254        1
20545       1
16949       1
20521       1
12364       1
21334       1
9054        1
20524       1
8921        1
21330       1
20526       1
9248        1
9043        1
17981       1
20547       1
9042        1
17803       1
21551       1
18267       1
16606       1
Name: Dealer, Length: 8386, dtype: int64

In [69]:
len(df_ship["Dealer"].unique())

8386

This is more significant.  The left column is the ID number of a dealer. The right column is the number of occurances of that ID. The dealer with the highest number of occurances has 6,252.  That's quite a few.
In total there are 8,386 different Dealer ID numbers that appear.  

In [72]:
df_ship.isna().sum()

VIN                   0
Plant                 0
Dealer                0
Plant_Arrival_Time    0
dtype: int64

In [73]:
df_ship

VIN Plant  Dealer  \
0        000003CD-DD91-4EAF-8CDB-D31A35CB4C65    RO   14135   
1        00000686-CBF0-455B-BDA8-85EAADCC31B9    RO   13449   
2        000017BF-BDF6-4B0B-88D5-46823B40DC1B    RO   18672   
3        00001999-A906-4391-A4DC-D4AC235EA6A3    FF   13619   
4        00001EDD-CDE4-49EB-957A-6FB1FE093E80    RO   16070   
5        0000209E-135A-4EB7-B3CF-8B139BA3A24F    3A   13782   
6        0000253D-FB0F-4EB9-8FBB-F3B5F91C20C4    RO   16070   
7        000025D5-B3E3-40F2-B64E-FC0D92FE607A    3A   16263   
8        0000306A-CBB7-449C-A144-AC6EFEA1F8F3    FF   16017   
9        00003411-7D48-443E-93B9-47C9B6BABD05    SO   14629   
10       00003688-43CA-4A17-9C9C-386D131630C1    SO   13752   
11       00005A68-41A6-427E-ADF6-C56EAF93CF7B    3A   15716   
12       00006047-F9A6-4F79-9ADF-794057EB62E9    RO    9415   
13       00007349-8D8D-4372-AE48-8C39579BCDB4    RO   15265   
14       0000740E-C9F5-4E56-9F49-1CCAFAF0B6C1    SO   13914   
15       00007626-5DF1-40FB-9A5F-D06064A51320    RO   15801   
16       00008602-56A6-4B65-803C-2680FAD31EFB    RO   15534   
17       00008C21-6A2B-4890-8A97-3002652B60A3    3A   13196   
18       00008E79-F383-480C-9403-97DFEA64A593    SO   15279   
19       000097F2-E1B4-408E-BBDA-70577F9351AD    FF   14797   
20       00009DB2-8646-41F4-90DB-E446444A556D    RO   20369   
21       0000A295-4E7F-4BEE-80F5-941C78AFE491    FF   15535   
22       0000AE2C-DFFA-4CC8-B59D-C70E003FACDE    SO   15966   
23       0000B87E-5F06-48EA-90C1-128538F306A7    RO   14666   
24       0000D7F9-DCB5-4CEC-ADF1-3763367A5DF2    RO   13577   
25       0000DF31-1E60-404E-931D-403D0AA39673    3A   16214   
26       0000E5BB-5C79-4EF2-BACC-CB32864F8006    SO   20570   
27       0000E801-8446-4EE9-B60F-02F964F0D5DA    3A   13777   
28       0000EC58-BA13-424A-A63B-D7DF7A710C84    SO   20625   
29       000100F2-6A46-4F7A-8790-D7F0C8F2BE69    3A   20783   
...                                       ...   ...     ...   
1866657  FFFF52F6-708E-43E8-A270-1D48CB40A465    RO   16147   
1866658  FFFF5AAF-B213-4BC6-9E15-56EDD331F2AD    SO   19182   
1866659  FFFF5F8B-146E-463F-946C-0851A6B815DC    FF    8701   
1866660  FFFF615F-B106-41E8-90EF-30AC0AE9A5CC    3A   11991   
1866661  FFFF619C-C6C9-48B7-B139-AE018A4F436A    SO   19313   
1866662  FFFF6B78-E878-4891-85EE-5B97BA9A5215    SO   15956   
1866663  FFFF6DA1-3628-49D1-B65B-D9D9EE93481A    RO   15521   
1866664  FFFF6FEE-E2AC-4AD3-8CD1-E597A5D0354F    3A   17937   
1866665  FFFF7545-74C3-4F79-8CCB-6BA2E5677215    RO   13265   
1866666  FFFF79B6-D596-487F-AFFB-D22C1846AFF5    3A   15647   
1866667  FFFF7BBC-18B5-4493-A979-C1B428AE037A    3A   16760   
1866668  FFFF9105-C26E-4AC2-A782-2B34B490170A    FF   14604   
1866669  FFFF91B8-2D5A-446B-82B6-E08B82220AB2    FF   13220   
1866670  FFFF943A-694C-40AC-AAC3-F4AD14631B65    RO   18770   
1866671  FFFF9BEE-3C4D-406B-92B1-9A96F51347B0    RO   19244   
1866672  FFFF9CD7-62CD-418D-8DBD-12D7B989A3BE    RO   14518   
1866673  FFFF9D75-BE5E-4730-AFF2-7B997F0C9693    RO    9213   
1866674  FFFFA905-0DEC-4215-B2BB-09D9D339FBD3    RO   18674   
1866675  FFFFABC1-FEDB-462D-83E3-3C0F6D54FBF8    RO   16092   
1866676  FFFFAD80-181E-4BAC-8DF4-8865FAB5CEEA    SO   20710   
1866677  FFFFC36C-02DB-425F-8C74-873C4FA657F6    RO   15606   
1866678  FFFFC3B4-A93E-43AC-BF83-83383B973681    SO   19412   
1866679  FFFFC5F9-A605-464B-ACE0-083146B17AE3    3A   20891   
1866680  FFFFC95D-D212-4E19-9A69-8942FEF253CC    SO   13319   
1866681  FFFFCBB8-4E30-4898-8FBB-85C3E9A11207    FF   14716   
1866682  FFFFDA87-FB19-4C91-9F50-623F58377F92    SO   13788   
1866683  FFFFDC17-10BB-42C5-AD72-FEAD19629F73    3A   19973   
1866684  FFFFDD42-52C3-4B23-B7E9-33F464374949    3A   13132   
1866685  FFFFE7FF-AA89-450F-97F6-BCFC6D28508E    FF   19962   
1866686  FFFFF4F1-65C3-4F86-89CC-5269053DF9BA    RO   19274   

          Plant_Arrival_Time  
0           3/8/2015 1:53:00  
1          3/16/2015 2:58:00  
2          3/31/2015 6:17:00  
3        10

### Input Cost and Location

In [74]:
df_inpu = pd.read_excel("GM Datasets/Input_Cost%2C+Location.xlsx")

In [75]:
len(df_inpu)

21723

In [76]:
df_inpu

Location  Dealer?  Latitude  Longitude
0            2        1     36.80    -158.91
1            3        1     21.30    -174.80
2            4        1     17.71     166.80
3            5        1     26.77     167.85
4            6        1     27.32    -169.05
5            7        1     21.45     163.64
6            8        1     25.64    -179.47
7            9        1     19.65     178.40
8           10        1     26.36     162.74
9           11        1     35.92    -165.01
10          12        1     20.15     162.08
11          13        1     19.92     170.00
12          14        1     25.53     165.74
13          15        1     23.41     174.55
14          16        1     33.30    -155.58
15          17        1     20.58     172.07
16          18        1     22.49     175.35
17          19        1     27.16     174.34
18          20        1     17.64     158.74
19          21        1     19.23    -177.25
20          22        1     25.91     174.97
21          23        1     43.09    -174.95
22          24        1     27.14     167.57
23          25        1     29.55    -178.90
24          26        1     27.27     178.90
25          27        1     13.01     158.89
26          28        1     20.82     166.16
27          29        1     21.22     176.45
28          30        1     19.99    -178.43
29          31        1     25.84     174.72
...        ...      ...       ...        ...
21693    21695        1     34.75     163.87
21694    21696        1     19.97     170.61
21695    21697        1     19.97     170.61
21696    21698        1     19.97     170.61
21697    21699        1     19.97     170.61
21698    21700        1     19.97     170.61
21699    21701        1     20.04     170.70
21700    21702        1     35.74    -156.45
21701    21703        1     35.89    -157.81
21702    21704        1     17.84     158.97
21703    21705        1     18.78     159.24
21704    21706        1     29.34     171.57
21705    21707        1     15.48     161.11
21706    21708        1     21.11     162.09
21707    21709        1     27.59     170.38
21708    21710        1     22.49     169.46
21709    21711        1     30.61     166.11
21710    21712        1     21.32     172.68
21711    21713        1     21.46     164.23
21712    21714        1     13.97     155.10
21713    21715        1     20.96     163.29
21714    21716        1     31.47     166.08
21715    21717        1     21.70     162.90
21716    21718        1     33.23     165.35
21717    21719        1     22.70     171.77
21718    21720        1     19.78     160.99
21719    21721        1     22.26     163.73
21720    21722        1     21.75     172.47
21721    21723        1     21.32     172.67
21722    21724        1     20.61     167.18

[21723 rows x 4 columns]

In [77]:
df_ship.isna().sum()

VIN                   0
Plant                 0
Dealer                0
Plant_Arrival_Time    0
dtype: int64

No data missing

## Reference Datasets

In [78]:
os.listdir("Reference Datasets/")

['DataSet1',
 'DataSet1.zip',
 'DataSet2',
 'DataSet2.zip',
 'Sample_Output+Summary.xlsx']

In [79]:
ref_summary = pd.read_excel("Reference Datasets/Sample_Output+Summary.xlsx")
ref_summary

Total Cost ($)  Cost/Veh ($)  Logistics Cost \n($)  \
Dataset1    1.486253e+09    829.644123          4.473317e+08   
Dataset2    1.593282e+09    806.963104          5.055292e+08   

          Logistics Cost \n($) Milkun  VDC Cost\n($)   \
Dataset1                 6.014917e+08       199972408   
Dataset2                 5.979171e+08       224482924   

          Lead Time Cost\n($) 10 per day  VDCs Added \n(#)  \
Dataset1                       237456879                 0   
Dataset2                       265353260                 0   

          Lead Time Avg \n(Day)  Lead Time St Dev \n(Day)  
Dataset1              13.255077                 17.512586  
Dataset2              13.423299                  8.696794

It appears to literally be a summary of the two datasets.  Alright, lets dig in a bit.  

### Dataset 1

In [80]:
os.listdir("Reference Datasets/Dataset1")

['Results Template 1_VDC_DataSet1.csv',
 'Results Template 3_RoutingDetails_DataSet1.csv',
 'Results Template 4_ShipDetails_DataSet1.csv',
 'Results Template 5_ShipStops_DataSet1.csv',
 'Results Template 6_Leadtime_DataSet1.csv',
 'VehicleShipmentRequirement_DataSet1.csv']

Ok, few in here...

In [81]:
res_1 = pd.read_csv("Reference Datasets/Dataset1/Results Template 1_VDC_DataSet1.csv")
res_1

VDC    Lat    Long  Capacity (veh)  VDC Opening Cost($)  VDC Vehicles  \
0   3A  32.47 -178.36            4730                    0        464121   
1   3F  22.51  172.08             170                    0          1056   
2   4J  21.89  163.13             840                    0         62109   
3   7J  21.92  169.40             480                    0         10121   
4   7M  20.16  170.06             200                    0          5002   
5   BC  29.49  171.75            1150                    0         89753   
6   BE  17.85  159.17            1710                    0        114387   
7   CE  21.86  174.25             810                    0         47980   
8   DI  27.88  166.21             830                    0         51626   
9   DO  26.22  177.54            1080                    0         75584   
10  DV  20.24  171.74              50                    0           836   
11  DW  21.90 -149.43             450                    0         33681   
12  DZ  33.43  175.40            1690                    0        125189   
13  EC  17.86  166.57             270                    0         14365   
14  FF  19.57  170.72            4590                    0        280244   
15  GU  28.58  175.44             900                    0         58491   
16  JC  31.52  166.04             720                    0         47750   
17  LM  35.55 -158.13            1180                    0         73069   
18  MN  19.74 -179.85            1670                    0         87896   
19  MR  14.98  161.35             380                    0         22105   
20  NM  34.92 -163.57             460                    0         30959   
21  NZ  20.32  170.25             900                    0         85716   
22  OX  29.86 -177.36             720                    0         46861   
23  PB  23.87  163.34             330                    0         22714   
24  PH  33.11 -153.12             530                    0         33575   
25  QT  35.97 -177.69             730                    0         43875   
26  RO  21.72  171.69            5310                    0        563942   
27  RS  27.73 -162.19             440                    0         33119   
28  RX  19.83  170.24              20                    0           992   
29  SO  44.82 -176.64            5020                    0        556144   
30  SU  22.16 -155.32             170                    0         12709   
31  SZ  31.89 -170.47             440                    0         24436   
32  UL  20.76  167.55             790                    0         49920   
33  VE  15.52 -158.06             800                    0         51641   
34  VG  16.73 -167.25             230                    0         15822   
35  QW  19.90  169.95             200                    0           109   
36  WH  13.69  151.40             200                    0         16622   
37  WK  20.10  169.93             200                    0         18951   

    Overflow Vehicles  Overflow Vehicle*Days  Tota Cost($) From Time  \
0                 603                    603      26679452  1/1/2011   
1                   0                      0        176900  1/1/2011   
2                5610                   9986       3926894  1/1/2011   
3                   0                      0        856450  1/1/2011   
4                   0                      0        396100  1/1/2011   
5                8380                  18357       5651978  1/1/2011   
6               11202                  23955       7399530  1/1/2011   
7                1495                   3615       3049610  1/1/2011   
8                2385                   2858       3270182  1/1/2011   
9                8403                  14700       4878490  1/1/2011   
10                  0                      0         78300  1/1/2011   
11               3925                   6401       2155904  1/1/2011   
12              19202                  34696       8207994  1/1/2011   
13               2172         

In [82]:
res_3 = pd.read_csv("Reference Datasets/Dataset1/Results Template 3_RoutingDetails_DataSet1.csv")
res_3

VIN    LOC          Arrive Time          Depart Time Depart Mode  \
0              1     3A  12/14/2011 21:42:00   12/15/2011 7:52:00           R   
1              1     BE  12/23/2011 22:43:00     1/1/2012 6:00:00           T   
2              1  15863     1/2/2012 0:22:00     1/2/2012 0:22:00           T   
3              2     RO     3/8/2012 1:53:00    3/15/2012 6:00:00           T   
4              2  14135   3/19/2012 11:26:00   3/19/2012 11:26:00           T   
5              3     RO    3/16/2012 2:58:00    3/16/2012 8:01:00           R   
6              3     OX   3/20/2012 17:50:00    3/25/2012 6:00:00           T   
7              3  13449   3/25/2012 18:29:00   3/25/2012 18:29:00           T   
8              4     RO    3/31/2012 6:17:00    4/1/2012 17:09:00           R   
9              4     WH    4/8/2012 22:24:00    4/14/2012 6:00:00           T   
10             4  18672   4/15/2012 14:35:00   4/15/2012 14:35:00           T   
11             5     FF  10/18/2012 16:58:00   10/19/2012 8:01:00           R   
12             5     PH   10/31/2012 9:17:00    11/7/2012 6:00:00           T   
13             5  13619   11/7/2012 14:15:00   11/7/2012 14:15:00           T   
14             6     SO   9/13/2011 20:33:00    9/14/2011 3:37:00           R   
15             6     QT    9/17/2011 5:17:00    9/23/2011 6:00:00           T   
16             6  15654   9/23/2011 11:20:00   9/23/2011 11:20:00           T   
17             7     RO   10/19/2012 4:35:00   10/22/2012 6:00:00           T   
18             7  16070  10/22/2012 23:45:00  10/22/2012 23:45:00           T   
19             8     3A    12/8/2012 8:46:00   12/9/2012 19:26:00           R   
20             8     JC   12/14/2012 9:16:00     1/5/2013 6:00:00           T   
21             8  16263     1/6/2013 4:09:00     1/6/2013 4:09:00           T   
22             9     RO   9/28/2011 14:46:00    9/29/2011 7:12:00           R   
23             9     MN    10/2/2011 2:46:00    10/7/2011 6:00:00           T   
24             9  13463   10/8/2011 23:41:00   10/8/2011 23:41:00           T   
25            10     FF    5/22/2012 3:01:00   5/22/2012 16:28:00           R   
26            10     DO   5/25/2012 20:15:00     6/8/2012 6:00:00           T   
27            10  16017    6/10/2012 2:17:00    6/10/2012 2:17:00           T   
28            11     SO  12/14/2012 22:16:00  12/16/2012 15:09:00           R   
29            11     4J   12/26/2012 9:50:00     1/4/2013 6:00:00           T   
...          ...    ...                  ...                  ...         ...   
5064877  1791431     RO   3/21/2012 12:20:00   3/21/2012 19:25:00           R   
5064878  1791431     BE    3/26/2012 2:09:00     4/2/2012 6:00:00           T   
5064879  1791431  19244    4/2/2012 14:02:00    4/2/2012 14:02:00           T   
5064880  1791432     RO    8/12/2012 0:07:00    8/12/2012 5:21:00           R   
5064881  1791432     4J   8/14/2012 23:11:00    8/24/2012 6:00:00           T   
5064882  1791432  14518    8/25/2012 2:09:00    8/25/2012 2:09:00           T   
5064883  1791433     RO     6/8/2012 4:22:00   6/11/2012 22:57:00           R   
5064884  1791433     WH    6/19/2012 4:12:00    6/27/2012 6:00:00           T   
5064885  1791433  18674   6/28/2012 14:39:00   6/28/2012 14:39:00           T   
5064886  1791434     SO   4/27/2012 13:18:00    4/28/2012 8:48:00           R   
5064887  1791434     GU     5/4/2012 9:10:00    5/10/2012 6:00:00           T   
5064888  1791434  19412   5/10/2012 20:31:00   5/10/2012 20:31:00           T   
5064889  1791435     3A    5/31/2012 8:34:00     6/6/2012 6:00:00           T   
5064890  1791435  20891    6/7/2012 10:26:00    6/7/2012 10:26:00           T   
5064891  1791436     SO    2/29/2012 2:32:00   2/29/2012 13:19:00           R   
5064892  1791436     MN     3/9/2012 6:25:00    3/13/2012 6:00:00           T   
5064893  1791436  13319    3/14/2012 9:07:00    3/14/2012 9:07:00           T   
5064894  1791437     3A    5/6/2012 20:05:00

In [83]:
res_4 = pd.read_csv("Reference Datasets/Dataset1/Results Template 4_ShipDetails_DataSet1.csv")
res_4

Shipment ID  Load  Distance      Cost
0          1/1/2011-2409-MR-1     3    114.12    656.49
1          1/1/2011-5705-MR-1    10   3779.36  15317.47
2          1/1/2011-5705-MR-2     3   7301.83  29407.33
3          1/1/2011-6164-MR-1     3    268.71   1274.84
4          1/1/2011-6684-MR-1     4   1239.86   5159.47
5          1/1/2011-7643-MR-1    10    223.98   1095.95
6          1/1/2011-7643-MR-2     5    108.20    632.81
7          1/1/2011-7656-MR-1     4    115.62    662.49
8          1/1/2011-7665-MR-1     2      2.75    211.00
9          1/1/2011-7671-MR-1     1     37.45    349.81
10      1/1/2012-1770-13747-1    10     85.41    541.64
11      1/1/2012-1770-13783-1    10     96.01    584.06
12      1/1/2012-1770-20604-1    10    196.92    987.69
13         1/1/2012-1770-MR-1    10     61.72    446.88
14        1/1/2012-1770-MR-10    10    383.75   1735.01
15        1/1/2012-1770-MR-11    10    177.03    908.13
16        1/1/2012-1770-MR-12    10     61.52    446.09
17        1/1/2012-1770-MR-13    10   1904.81   7819.26
18        1/1/2012-1770-MR-14     3     97.68    590.75
19         1/1/2012-1770-MR-2    10    137.12    748.48
20         1/1/2012-1770-MR-3    10    457.10   2028.42
21         1/1/2012-1770-MR-4     8    196.92    987.69
22         1/1/2012-1770-MR-5     2    270.69   1282.76
23         1/1/2012-1770-MR-6    10    757.67   3230.70
24         1/1/2012-1770-MR-7    10    218.82   1075.31
25         1/1/2012-1770-MR-8     7    643.06   2772.24
26         1/1/2012-1770-MR-9     3     96.01    584.06
27         1/1/2012-2409-MR-1    10    360.81   1643.27
28         1/1/2012-2409-MR-2    10    403.99   1815.96
29         1/1/2012-2409-MR-3    10    537.10   2348.43
...                       ...   ...       ...       ...
200776   9/9/2012-7671-8701-1    10    101.02    604.08
200777     9/9/2012-7671-MR-1    10    453.16   2012.67
200778     9/9/2012-7671-MR-2     6    461.54   2046.17
200779     9/9/2012-7671-MR-3     4    101.02    604.08
200780     9/9/2012-7671-MR-4     7    166.36    865.46
200781     9/9/2012-7672-MR-1    10    460.59   2042.38
200782     9/9/2012-7672-MR-2     8    560.12   2440.49
200783     9/9/2012-7676-MR-1    10   1975.56   8102.24
200784     9/9/2012-7676-MR-2    10    229.14   1116.58
200785     9/9/2012-7676-MR-3    10    348.04   1592.16
200786     9/9/2012-7676-MR-4    10    587.87   2551.49
200787     9/9/2012-7676-MR-5    10    507.17   2228.69
200788     9/9/2012-7676-MR-6    10    739.29   3157.17
200789     9/9/2012-7676-MR-7     8    549.49   2397.99
200790     9/9/2012-7679-MR-1    10   1152.50   4810.01
200791     9/9/2012-7679-MR-2    10   2374.00   9696.00
200792     9/9/2012-7679-MR-3    10    474.94   2099.77
200793     9/9/2012-7679-MR-4    10    574.16   2496.67
200794     9/9/2012-7679-MR-5    10    235.77   1143.08
200795     9/9/2012-7679-MR-6    10    680.25   2921.01
200796     9/9/2012-7679-MR-7    10    714.33   3057.32
200797     9/9/2012-7679-MR-8    10    298.68   1394.75
200798     9/9/2012-7679-MR-9     5     14.19    256.79
200799     9/9/2012-7680-MR-1    10    593.56   2574.24
200800     9/9/2012-7680-MR-2     4    166.00    864.01
200801     9/9/2012-7689-MR-1    10   1202.94   5011.77
200802     9/9/2012-7689-MR-2    10    609.34   2637.36
200803     9/9/2012-7689-MR-3     3    386.05   1744.21
200804     9/9/2012-8461-MR-1    10   1250.53   5202.13
200805     9/9/2012-8461-MR-2     4   1068.44   4473.79

[200806 rows x 4 columns]

In [84]:
res_5 = pd.read_csv("Reference Datasets/Dataset1/Results Template 5_ShipStops_DataSet1.csv")
res_5

Shipment ID Location  Sequence      Departure Time  \
0        1/1/2011-2409-MR-1       4J         1    1/1/2011 6:00:00   
1        1/1/2011-2409-MR-1     9089         2   1/1/2011 11:48:00   
2        1/1/2011-5705-MR-1       BE         1    1/1/2011 6:00:00   
3        1/1/2011-5705-MR-1     9092         2   1/1/2011 11:28:00   
4        1/1/2011-5705-MR-1      660         3   1/6/2011 12:41:00   
5        1/1/2011-5705-MR-1     3820         4   1/6/2011 17:57:00   
6        1/1/2011-5705-MR-2       BE         1    1/1/2011 6:00:00   
7        1/1/2011-5705-MR-2     3820         2    1/6/2011 8:48:00   
8        1/1/2011-5705-MR-2     9086         3   1/11/2011 6:28:00   
9        1/1/2011-5705-MR-2     9088         4  1/11/2011 15:22:00   
10       1/1/2011-6164-MR-1       CE         1    1/1/2011 6:00:00   
11       1/1/2011-6164-MR-1     8779         2   1/1/2011 12:10:00   
12       1/1/2011-6164-MR-1     8778         3   1/1/2011 18:56:00   
13       1/1/2011-6684-MR-1       DO         1    1/1/2011 6:00:00   
14       1/1/2011-6684-MR-1    14061         2    1/3/2011 1:19:00   
15       1/1/2011-7643-MR-1       LM         1    1/1/2011 6:00:00   
16       1/1/2011-7643-MR-1     8631         2   1/1/2011 10:06:00   
17       1/1/2011-7643-MR-1     8636         3   1/1/2011 12:49:00   
18       1/1/2011-7643-MR-1     8634         4   1/1/2011 15:09:00   
19       1/1/2011-7643-MR-1     8638         5   1/1/2011 17:32:00   
20       1/1/2011-7643-MR-1     8632         6   1/1/2011 23:25:00   
21       1/1/2011-7643-MR-2       LM         1    1/1/2011 6:00:00   
22       1/1/2011-7643-MR-2     8632         2   1/1/2011 11:20:00   
23       1/1/2011-7643-MR-2     8639         3   1/1/2011 13:36:00   
24       1/1/2011-7656-MR-1       PB         1    1/1/2011 6:00:00   
25       1/1/2011-7656-MR-1     9121         2   1/1/2011 11:51:00   
26       1/1/2011-7665-MR-1       RS         1    1/1/2011 6:00:00   
27       1/1/2011-7665-MR-1     8700         2    1/1/2011 8:05:00   
28       1/1/2011-7671-MR-1       SU         1    1/1/2011 6:00:00   
29       1/1/2011-7671-MR-1     8699         2    1/1/2011 9:14:00   
...                     ...      ...       ...                 ...   
1067507  9/9/2012-7680-MR-1    18546         3   9/10/2012 0:02:00   
1067508  9/9/2012-7680-MR-1    18560         4   9/10/2012 5:07:00   
1067509  9/9/2012-7680-MR-1    18531         5   9/10/2012 9:46:00   
1067510  9/9/2012-7680-MR-2       VG         1    9/9/2012 6:00:00   
1067511  9/9/2012-7680-MR-2    18531         2   9/9/2012 13:32:00   
1067512  9/9/2012-7689-MR-1       WK         1    9/9/2012 6:00:00   
1067513  9/9/2012-7689-MR-1    18799         2   9/9/2012 21:17:00   
1067514  9/9/2012-7689-MR-1    18579         3  9/10/2012 12:42:00   
1067515  9/9/2012-7689-MR-1    18568         4  9/10/2012 15:26:00   
1067516  9/9/2012-7689-MR-1    18593         5   9/11/2012 0:16:00   
1067517  9/9/2012-7689-MR-1    18570         6   9/11/2012 3:50:00   
1067518  9/9/2012-7689-MR-1    18588         7  9/11/2012 10:03:00   
1067519  9/9/2012-7689-MR-2       WK         1    9/9/2012 6:00:00   
1067520  9/9/2012-7689-MR-2    18588         2   9/9/2012 11:33:00   
1067521  9/9/2012-7689-MR-2    18573         3   9/9/2012 18:01:00   
1067522  9/9/2012-7689-MR-2    18577         4   9/9/2012 20:37:00   
1067523  9/9/2012-7689-MR-2    18595         5   9/10/2012 1:50:00   
1067524  9/9/2012-7689-MR-2    18581         6   9/10/2012 6:35:00   
1067525  9/9/2012-7689-MR-2    18746         7  9/10/2012 14:15:00   
1067526  9/9/2012-7689-MR-3       WK         1    9/9/2012 6:00:00   
1067527  9/9/2012-7689-MR-3    18787         2   9/9/2012 20:52:00   
1067528  9/9/2012-8461-MR-1       WH         1    9/9/2012 6:00:00   
1067529  9/9/2012-8461-MR-1    18676         2   9/9/2012 17:57:00   
1067530  9/9/2012-8461-MR-1    18678         3   9/9/2012 21:45:00   
1067531  9/9/2012-8461-MR-1    18648         4   9/10/2012 7:34:00   
1067532  9/9/2012-8461-MR-

In [85]:
res_6 = pd.read_csv("Reference Datasets/Dataset1/Results Template 6_Leadtime_DataSet1.csv")
res_6

VIN_ID  Lead Time(Day)
0              1           18.11
1              2           11.39
2              3            9.64
3              4           15.34
4              5           19.88
5              6            9.61
6              7            3.79
7              8           28.80
8              9           10.37
9             10           18.96
10            11           21.31
11            12            2.45
12            13           20.77
13            14           14.71
14            15           16.45
15            16           11.26
16            17           17.74
17            18           10.75
18            19            8.80
19            20           27.85
20            21           22.89
21            22           11.07
22            23           12.17
23            24           13.75
24            25            7.45
25            26            7.53
26            27            9.63
27            28            3.71
28            29            3.21
29            30           14.00
...          ...             ...
1791411  1791412           12.06
1791412  1791413           10.13
1791413  1791414           27.75
1791414  1791415            7.17
1791415  1791416           18.39
1791416  1791417           17.07
1791417  1791418           17.72
1791418  1791419            4.04
1791419  1791420           13.81
1791420  1791421           14.43
1791421  1791422           18.56
1791422  1791423            2.67
1791423  1791424           29.01
1791424  1791425           16.44
1791425  1791426           10.39
1791426  1791427            4.94
1791427  1791428           12.83
1791428  1791429           12.02
1791429  1791430            6.47
1791430  1791431           12.07
1791431  1791432           13.08
1791432  1791433           20.42
1791433  1791434           13.30
1791434  1791435            7.07
1791435  1791436           14.27
1791436  1791437           15.07
1791437  1791438           13.13
1791438  1791439            8.07
1791439  1791440            9.33
1791440  1791441           17.30

[1791441 rows x 2 columns]

In [86]:
res_veh = pd.read_csv("Reference Datasets/Dataset1/VehicleShipmentRequirement_DataSet1.csv")
res_veh

VIN Plant  Dealer   Plant_Arrival_Time
0              1    3A   15863  12/14/2011 21:42:00
1              2    RO   14135     3/8/2012 1:53:00
2              3    RO   13449    3/16/2012 2:58:00
3              4    RO   18672    3/31/2012 6:17:00
4              5    FF   13619  10/18/2012 16:58:00
5              6    SO   15654   9/13/2011 20:33:00
6              7    RO   16070   10/19/2012 4:35:00
7              8    3A   16263    12/8/2012 8:46:00
8              9    RO   13463   9/28/2011 14:46:00
9             10    FF   16017    5/22/2012 3:01:00
10            11    SO   14629  12/14/2012 22:16:00
11            12    FF   16548   8/26/2011 11:24:00
12            13    RO   14404   11/19/2011 6:19:00
13            14    FF   21716   4/26/2011 19:11:00
14            15    3A   19553    9/30/2011 7:34:00
15            16    3A   15716    2/11/2012 5:33:00
16            17    3A   21611     9/5/2011 0:17:00
17            18    RO   14852   2/26/2011 21:08:00
18            19    RO    9415   6/11/2012 22:34:00
19            20    RO    8982   4/30/2011 19:12:00
20            21    FF   16016     2/9/2011 3:51:00
21            22    RO   15265     3/1/2012 9:33:00
22            23    SO   13914   7/25/2012 14:50:00
23            24    SO   13449    2/4/2011 14:10:00
24            25    SO   15279   3/31/2012 21:10:00
25            26    FF   14797    6/8/2012 23:52:00
26            27    RO   14676    4/30/2011 3:02:00
27            28    FF   15535    6/6/2012 20:49:00
28            29    FF   16496   4/12/2011 19:32:00
29            30    SO   15966    9/23/2012 8:20:00
...          ...   ...     ...                  ...
1791401  1791412    3A   15604   10/17/2011 5:15:00
1791402  1791413    FF   14734   9/18/2011 18:45:00
1791403  1791414    SO   15513  11/25/2012 15:07:00
1791404  1791415    SO   13785   7/22/2012 22:58:00
1791405  1791416    SO   13713   12/24/2012 5:14:00
1791406  1791417    3A   16788  10/11/2011 18:59:00
1791407  1791418    RO   14645   4/27/2011 17:10:00
1791408  1791419    RO   19237    9/19/2012 1:36:00
1791409  1791420    FF   13691   4/23/2011 17:48:00
1791410  1791421    FF    9451  11/23/2012 14:34:00
1791411  1791422    FF    8604   4/25/2011 15:30:00
1791412  1791423    RO   16147     2/3/2012 0:36:00
1791413  1791424    FF    8701    6/27/2012 8:03:00
1791414  1791425    RO   15521   6/19/2012 12:23:00
1791415  1791426    RO   13265   3/10/2012 22:18:00
1791416  1791427    3A   15647    3/5/2012 22:57:00
1791417  1791428    RO   14914    4/17/2011 1:23:00
1791418  1791429    FF   14604  12/23/2012 21:14:00
1791419  1791430    FF   13220    3/24/2012 8:11:00
1791420  1791431    RO   19244   3/21/2012 12:20:00
1791421  1791432    RO   14518    8/12/2012 0:07:00
1791422  1791433    RO   18674     6/8/2012 4:22:00
1791423  1791434    SO   19412   4/27/2012 13:18:00
1791424  1791435    3A   20891    5/31/2012 8:34:00
1791425  1791436    SO   13319    2/29/2012 2:32:00
1791426  1791437    3A   19973    5/6/2012 20:05:00
1791427  1791438    RO   17147   11/4/2011 21:35:00
1791428  1791439    3A   13813   3/30/2011 22:41:00
1791429  1791440    RO   19274   2/19/2012 20:48:00
1791430  1791441    SO   15238    9/20/2011 3:14:00

[1791431 rows x 4 columns]

### Dataset 2

In [87]:
os.listdir("Reference Datasets/Dataset2")

['Results Template 1_VDC_DataSet2.csv',
 'Results Template 3_RoutingDetails_DataSet2.csv',
 'Results Template 4_ShipDetails_DataSet2.csv',
 'Results Template 5_ShipStops_DataSet2.csv',
 'Results Template 6_Leadtime_DataSet2.csv',
 'VehicleShipmentRequirement_DataSet2.csv']

I think we can safely assume this dataset will be similar to the first one, just with different values, etc.
I think we've got our snapshot.